cmd/options.go

```diff
@@ -254,11 +253,51 @@ func (c *snapshotterOptions) validate() error {
 	if err := c.compressionConfig.Validate(); err != nil {
 		return err
 	}
-
 	return c.etcdConnectionConfig.Validate()
 }
 
 // complete completes the config.
 func (c *snapshotterOptions) complete() {
 	c.snapstoreConfig.Complete()
 }
+
+type copierOptions struct {
+	sourceSnapStoreConfig *types.SnapstoreConfig
+	snapstoreConfig       *types.SnapstoreConfig
+	maxSnapshots          int
+	maxSnapshotAge        int
+}
+
+func newCopierOptions() *copierOptions {
+	sourceSnapStoreConfig := snapstore.NewSnapstoreConfig()
+	sourceSnapStoreConfig.IsSource = true
+	return &copierOptions{
+		sourceSnapStoreConfig: sourceSnapStoreConfig,
+		snapstoreConfig:       snapstore.NewSnapstoreConfig(),
+	}
+}
+
+func (c *copierOptions) addFlags(fs *flag.FlagSet) {
+	fs.IntVar(&c.maxSnapshots, "max-snapshots", -1, "copy the last N snapshots form the store sorted by date from newest to oldest")
+	fs.IntVar(&c.maxSnapshotAge, "max-snapshot-age", -1, "copy the last snapshots form the store not older than N days")
+	c.sourceSnapStoreConfig.AddSourceFlags(fs)
+	c.snapstoreConfig.AddFlags(fs)
+}
+
+func (c *copierOptions) validate() error {
+	if err := c.snapstoreConfig.Validate(); err != nil {
+		return err
+	}
+	return c.sourceSnapStoreConfig.Validate()
+}
+
+func (c *copierOptions) complete() {
+	if c.maxSnapshots <= 0 {
+		c.maxSnapshots = -1
+	}
+	if c.maxSnapshotAge <= 0 {
+		c.maxSnapshotAge = -1
+	}
```

Comment:
Instead of this, we should validate `maxSnapshots` and `maxSnapshotAge` for being >= -1 in `validate`.